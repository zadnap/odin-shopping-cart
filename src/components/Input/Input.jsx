import styles from './Input.module.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch } from '@fortawesome/free-solid-svg-icons';

function Input({
  id,
  icon = faSearch,
  type = 'text',
  name,
  value,
  placeholder,
  onChange,
  ...props
}) {
  return (
    <div className={styles.inputContainer}>
      <FontAwesomeIcon icon={icon} className={styles.icon} />
      <input
        id={id}
        type={type}
        className={styles.input}
        value={value}
        name={name}
        placeholder={placeholder}
        onChange={onChange}
        {...props}
      />
    </div>
  );
}

export default Input;
