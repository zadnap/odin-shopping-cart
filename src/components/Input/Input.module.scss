@use '@/styles/variables.scss' as *;

.inputContainer {
  display: flex;
  align-items: center;
  gap: 12px;
  background-color: $subbackground-color;
  color: $text-color;
  border-radius: 5px;
  padding: 7px 12px;

  .icon {
    color: $subtext-color;
  }

  .input {
    font-size: 1.5rem;
    color: inherit;
    background-color: transparent;
    border: none;
    outline: none;
    flex: 1;

    &::placeholder {
      color: $subtext-color;
    }
  }
}
