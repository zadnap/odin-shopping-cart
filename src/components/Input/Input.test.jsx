import { vi, describe, expect, it, beforeAll } from 'vitest';
import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Input from './Input';

beforeAll(() => {
  vi.mock('@fortawesome/react-fontawesome', () => ({
    FontAwesomeIcon: () => <span data-testid="fa-icon" />,
  }));
});

describe('Input component', () => {
  it('should render to match snapshot', () => {
    const { container } = render(
      <Input type="text" value="Hi" onChange={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });

  it('should call onChange handler when typed in', async () => {
    const onChange = vi.fn();
    const user = userEvent.setup();
    render(<Input onChange={onChange} />);

    const input = screen.getByRole('textbox');
    await user.type(input, 'Hi');

    expect(onChange).toHaveBeenCalled();
  });
});
