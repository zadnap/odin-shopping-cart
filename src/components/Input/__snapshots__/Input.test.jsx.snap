// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input component > should render to match snapshot 1`] = `
<div>
  <div
    class="_inputContainer_9af53b"
  >
    <span
      data-testid="fa-icon"
    />
    <input
      class="_input_9af53b"
      type="text"
      value="Hi"
    />
  </div>
</div>
`;
